Bug fixs: sql mode

xy720 opened the pull request:
Issue #2365 
This commit fixs the bug below,

FE throws a unexpected exception when encounter a query like :
Set sql_mode = '0,PIPES_AS_CONCAT'.

and make some change to sql mode analyze process, now the analyze process is no longer put  in SetVar.class, but in VariableMgr.class.

Commit:
commit 1: fix sql mode bugs

Commit:
commit 1: fix sql mode bugs

Commit:
commit 2: fix bugs

Commit:
commit 2: fix bugs

Commit:
commit 3: fix bugs

Commit:
commit 3: fix bugs

Commit:
commit 4: fix bugs

Commit:
commit 4: fix bugs

Commit:
commit 5: fix bugs

Commit:
commit 5: fix bugs

Commit:
commit 6: add expand

Commit:
commit 6: add expand

Commit:
add unit test

Commit:
add unit test

morningman commented on fe/src/main/java/org/apache/doris/qe/SqlModeHelper.java:
```diff
@@ -139,21 +143,26 @@ public static String decode(Long sqlMode) throws AnalysisException {
         for (String key : names) {
+            if (StringUtils.isNumeric(key)) {
+                value |= expand(Long.valueOf(key));
+                LOG.warn("Set sql mode with numerical value");
```
Why here is a warning log?
Here should only has debug level log.

morningman commented on fe/src/main/java/org/apache/doris/qe/VariableVarConverters.java:
```diff
@@ -0,0 +1,46 @@
+import org.apache.doris.common.DdlException;
+
+import java.util.Map;
+
```
Add comment to this class.
What is this class for, and how to add new converter?

Commit:
commit 8: add some comment

Commit:
commit 8: add some comment

morningman reviewed (dismissed):
LGTM

morningman committed:
Fixed a bug that Load job's state is incorrect when upgrading from 0.10.x to 0.11.x (#2356)

There is bug in Doris version 0.10.x. When a load job in PENDING or LOADING
state was replayed from image (not through the edit log), we forgot to add
the corresponding callback id in the CallbackFactory. As a result, the
subsequent finish txn edit logs cannot properly finish the job during the
replay process. This results in that when the FE restarts, these load jobs
that should have been completed are re-entered into the pending state,
resulting in repeated submission load tasks.

Those wrong images are unrecoverable, so that we have to reset all load jobs
in PENDING or LOADING state when restarting FE, depends on its corresponding
txn's status, to avoid submit jobs repeatedly.

If corresponding txn exist, set load job' state depends on txn's status.
If txn does not exist, may be the txn has been removed due to label expiration.
So that we don't know the txn is aborted or visible. So we have to set the job's state
as UNKNOWN, which need handle it manually.

morningman committed:
Fixed a bug that HttpServer in unit test does not start correctly. (#2361)

Because the http client in unit test try to connect to the server when
server is not ready yet.

kangkaisen committed:
Add Bitmap index reader (#2319)

[STORAGE] [INDEX]

For #2061 and #2062

Add bitmap index reader
SegmentIterator support bitmap index
Add some metrics

morningman committed:
Add bdbje heartbeat timeout as a configuration of FE (#2366)

The timeline for this question is as follows:

1. For some reason, the master have lost contact with the other two followers.
Judging from the logs of the master, for almost 40 seconds, the master did not print any logs.
It is suspected that it is stuck due to full gc or other reasons, causing the
other two followers to think that the master has been disconnected.

2. After the other two followers re-elected, they continued to provide services.

3. The master node is manually restarted afterwards. When restarting it for the first time,
it needs to rollback some committed logs, so it needs to be closed and restarted again.
After restarting again, it returns to normal.

The main reason is that the master got stuck for 40 seconds for some reason.
This issue requires further observation.

At the same time, in order to alleviate this problem, we decided to set bdbje's heartbeat timeout
as a configurable value. The default is 30 seconds. Can be configured to 1 minute,
try to avoid this problem first.

wuyunfeng committed:
Optimize Doris On Elasticsearch performance (#2237)

Pure DocValue optimization for doris-on-es

Future todo:
Today, for every tuple scan we check if pure_docvalue is enabled, this is not reasonable,  should check pure_docvalue enabled for one whole scan outside,  I will add this todo in future

morningman committed:
Fixed bdbje heartbeat timeout config format bug (#2369)

The heartbeat config format should be like "30 s", not "30"
This CL is related to commit 261072ecdda7e8eb3ce685c557c6dab15488d1f3

Commit:
commit 9: rebase master
